drivers/metal/rendering_context_driver_metal.mm

```diff
@@ -172,7 +172,11 @@ void present(MDCommandBuffer *p_cmd_buffer) override final {
 		count--;
 		front = (front + 1) % frame_buffers.size();
 
-		[p_cmd_buffer->get_command_buffer() presentDrawable:drawable afterMinimumDuration:present_minimum_duration];
+		if (vsync_mode == DisplayServer::VSYNC_ENABLED) {
+			[p_cmd_buffer->get_command_buffer() presentDrawable:drawable afterMinimumDuration:present_minimum_duration];
```

Comment:
Note that `VSYNC_ADAPTIVE` and `VSYNC_MAILBOX` are treated as `VSYNC_ENABLED` on macOS, so I suggest checking for `vsync_mode != DisplayServer::VSYNC_DISABLED` instead.

Reply:
I don't think this is related but I can check

Reply:
Well I just triple-checked and this is confusing, but it does regress to unstable FPS as soon as this line is put into the conditional statement:

```
[p_cmd_buffer->get_command_buffer() presentDrawable:drawable afterMinimumDuration:present_minimum_duration];
```

I tried taking it out of it myself and recompiling and I get stable 120 as soon as I do that. I also did a check to make sure that the correct branch is running when it's in it, so it's not a problem with `vsync_mode` either

If there is some obscure metal-related cache that I may need to clear or something like that let me know because I don't know why this is happening

Reply:
Yeah I'm on 15.1 as well

https://github.com/user-attachments/assets/376306ae-b7bf-4758-9d9d-e9689a49c854

Reply:
I went to check the call stack but I can't reproduce the issue anymore no matter what I do. Apparently it was something external that caused different vsync values. In any case it appears to be unrelated to this PR, I'll open another issue if I face it again